Fix overlapping in Container Image summary screen

zakiva opened the pull request:
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1416185

zakiva commented:
@miq-bot add_label providers/containers, bug

zakiva commented:
@h-kataria Can you please review?
I changed the Envirement Varaiables table so it will not expand and cause an overlapping (another option is to switch places with one of the right side tables).
@simon3z We use an expanding table (with `_textual_multilabel.html.haml`) also in Containers and Container Builds summary screens, but I don't see an overlapping there with my current data. I wonder if we want to avoid an expanding there anyway.

simon3z commented:
> I changed the Environment Variables table so it will not expand and cause an overlapping (another option is to switch places with one of the right side tables).

@zakiva can you post a screenshot? I have the feeling that without the expansion then long variables will look too large (e.g. certificates, etc.)

zakiva commented:
> @zakiva can you post a screenshot? I have the feeling that without the expansion then long variables will look too large (e.g. certificates, etc.)

I'm adding 3 screenshots, each of them with a different zoom (ascending order): 

1)
![image1](https://cloud.githubusercontent.com/assets/11769555/22421552/6d6c8374-e6f0-11e6-9ed7-9ef63dcbfe69.png)

2)
![image2](https://cloud.githubusercontent.com/assets/11769555/22421550/6d683378-e6f0-11e6-8f52-f69dde13f0ff.png)

3)
![image3](https://cloud.githubusercontent.com/assets/11769555/22421551/6d6b39d8-e6f0-11e6-9d36-9e077254933c.png)

simon3z commented:
> I'm adding 3 screenshots, each of them with a different zoom (ascending order):

@zakiva yes bug is fixed but I wanted to check other places where `_textual_multilabel.html.haml` is used for its expansion capabilities (e.g. long env variables such as certificates, etc.)

See Alon's PR https://github.com/ManageIQ/manageiq/pull/6013

zakiva commented:
> @zakiva yes bug is fixed but I wanted to check other places where _textual_multilabel.html.haml is used for its expansion capabilities (e.g. long env variables such as certificates, etc.)

@simon3z I've changed the `max-width` limit so that we avoid overlapping and still support the expansion capabilities used in long env variables. 

Updated screenshot:

![image5](https://cloud.githubusercontent.com/assets/11769555/22423942/db5954fa-e6fc-11e6-8907-6c567ce9a7f6.png)

simon3z commented:
@zakiva where is the `200px` magic number coming from? How do we know it always looks good?

For example in this case:

![](https://cloud.githubusercontent.com/assets/11769555/22421551/6d6b39d8-e6f0-11e6-9d36-9e077254933c.png)

Would it be as large as the other tables or limited to `200px` therefore shorter in some cases?

zakiva commented:
> @zakiva where is the 200px magic number coming from? How do we know it always looks good?
> Would it be as large as the other tables or limited to 200px therefore shorter in some cases?

@simon3z Well I don't know where the previous number (350px) came from :)  the 200px limit is only for the `value` field, so the table will be as large as the other tables also in this case: 

![image7](https://cloud.githubusercontent.com/assets/11769555/22432029/4274bfca-e71c-11e6-9055-a3aa327e4a82.png)

This number is just the largest limit I could come with to ensure that there is no overlapping as described in the bug with my current data. However, I'm not certain this is the best way to guarantee it cc @h-kataria

simon3z commented:
> @simon3z Well I don't know where the previous number (350px) came from :)

@zakiva yes but we know it's not working, it's a wrong approach since it's based on a magic number and it's a bug we want to fix, so changing the value is not going to cut it (well unless we can explain the reason for that specific value).

> the 200px limit is only for the value field, so the table will be as large as the other tables also in this case:

That seems to work only for the case you presented there or the specific settings of your browser, or based on the other columns size. Why do you expect this to work on any browser? (E.g. why `200px` is the correct value?)

For example the magic number `200px` doesn't seem to work well enough in all cases as I was expecting:

![screenshot from 2017-01-30 23-28-43](https://cloud.githubusercontent.com/assets/130393/22444803/c64ddf16-e744-11e6-93fb-3d0e63bbbd23.png)


> This number is just the largest limit I could come with to ensure that there is no overlapping as described in the bug with my current data. However, I'm not certain this is the best way to guarantee it cc @h-kataria 

We need something that works more reliably. @h-kataria can you suggest a solution?

martinpovolny commented:
@simon3z : please, ping me when you are happy with this one. Thx!

simon3z commented:
> @simon3z : please, ping me when you are happy with this one. Thx!

@martinpovolny sure, still waiting for @h-kataria feedback

martinpovolny commented:
ping @h-kataria : please, take a look

zakiva commented:
@h-kataria can you please take a look?

h-kataria commented:
@epwinchell do you have any suggestions?

epwinchell commented:
@h-kataria I'll have a look. @simon3z @zakiva Please leave me a comment when there are styling issues going forward. Thanks.

epwinchell commented:
@zakiva Please rebase

simon3z commented:
> @h-kataria I'll have a look. @simon3z @zakiva Please leave me a comment when there are styling issues, going forward. Thanks.

@epwinchell sure, any suggestion here?

epwinchell commented:
@simon3z The branch needs to be be rebased

simon3z commented:
> @simon3z The branch needs to be rebased against master. Thanks

@epwinchell this PR is green now and there are no messages about a need for rebase.

![screenshot from 2017-02-24 17-07-24](https://cloud.githubusercontent.com/assets/130393/23310714/bf65d546-fab3-11e6-905b-3dfe556220b2.png)

Should we rebase because there's something we should see in master?

Also this is a one-liner currently:

```diff
-    max-width: 350px;
+    max-width: 200px;
```

Which we also believe is not the correct fix according to a [previous comment](https://github.com/ManageIQ/manageiq-ui-classic/pull/272#issuecomment-276215315).

So currently we're waiting on your suggestion on how to handle this.

epwinchell commented:
@simon3z  It's green, but It wont run with the current master in the Manageiq repo. 

`[----] F, [2017-02-24T09:59:53.013237 #79890:3fef915dcc10] FATAL -- : ActionView::Template::Error (couldn't find file 'excanvas' with type 'application/javascript'`

Please rebase. Thanks.

enoodle commented:
There seem to be a problem with displaying the environment variables due to some recent across the board changes in the UI. I am looking into this issue.

zakiva commented:
> @zakiva Please rebase

@epwinchell Rebased, thanks. 

@simon3z I now see that regardless of this BZ the Environment Variables table was moved to the right side of the page which fixes the overlapping (as I suggested above). 
I think we may consider this as a solution and close this PR:

![gifrecord_2017-02-28_105240](https://cloud.githubusercontent.com/assets/11769555/23398335/cfb03362-fda4-11e6-8aa9-1a4362b70871.gif)

simon3z commented:
@epwinchell @zakiva we don't care about the fact that the table at some resolutions is overflowing the page and activates the scrollbar?

epwinchell commented:
@simon3z I looked at this with @hayesr and we both agree there is no better solution without a full-redesign. I think this fix is adequate for the BZ.

epwinchell commented:
@miq-bot assign @martinpovolny

epwinchell commented:
@miq-bot add_label formatting

epwinchell commented:
@miq-bot add_label formatting/styling

simon3z commented:
> @simon3z I looked at this with @hayesr and we both agree there is no better solution without a full-redesign. I think this fix is adequate for the BZ.

@epwinchell @hayesr any plan on the future redesign that would properly fix this?

@zakiva also if the table was moved and there is no overlap anymore should we even bother to change the `max-width` from `350px;` to `200px;`?

zakiva commented:
> @zakiva also if the table was moved and there is no overlap anymore should we even bother to change the max-width from 350px; to 200px;?

@simon3z Well, changing the max-width will decrease the number of resolution levels that activate the scrollbar and maybe even not activate it at all (depending on the specific browser settings). 
Personally, I don't think this scrollbar is problematic, but if you don't like it then we should keep this change (otherwise, I'll close this PR).

epwinchell commented:
> @epwinchell @hayesr any plan on the future redesign that would properly fix this?

simon3z commented:
@epwinchell given that this is targeted to Euwe we have to properly identify the fix for the relevant BZ https://bugzilla.redhat.com/show_bug.cgi?id=1416185

IIUC the issue related to the overlap is not addressed here (as shown [in a previous comment](https://github.com/ManageIQ/manageiq-ui-classic/pull/272#issuecomment-276215315))

@epwinchell @zakiva can you reference the PR that moved the table on the right and actually avoided the overlap so that we can check if it's already targeted to Euwe or not?

epwinchell commented:
cc @dclarizio

simon3z commented:
Until we find out what to do I prefer to remove the euwe label since I don't want the process/BZ to continue with a PR that is not fixing the original issue.

@miq-bot rm_label euwe/yes

zakiva commented:
> @epwinchell @zakiva can you reference the PR that moved the table on the right and actually avoided the overlap so that we can check if it's already targeted to Euwe or not?

@simon3z I think that the table was moved in https://github.com/ManageIQ/manageiq-ui-classic/pull/274 which removed `_main.html.haml` partial files including `app/views/container_image/_main.html.haml` and added the tables in the controller instead. I see that this PR has label euwe/no. 
@martinpovolny Can you please confirm that?

martinpovolny commented:
I can confirm that I have removed the partials and created the `Textual{Group,Multilabel,...}` bits.

The styling should be done in a generic responsive way. It should not contain any container-specific formatting.

Looking forward we will serve just JSON with the properties probably throught the API and will do the formatting via a shared Angular component on the frontend.

So, please, no specific formatting or it will get broken again ;-)
